### PR TITLE
fix SetHooks don't work

### PR DESCRIPTION
**Pull request by yihuang**
Because `EthereumEventProcessor` have a copy of keeper instance, when origional keeper is modified, the copied one is out of date.

### CODE REVIEW

**Review by mvid — APPROVED** [review]

**Comment by mvid** [comment]
@jackzampolin @levicook is there a reason we have this EthereumEventProcessor type? Why isn't this just a keeper method?

File: module/x/gravity/keeper/keeper.go [hunk 1]
```diff
@@ -67,7 +67,6 @@ func NewKeeper(
 		PowerReduction: powerReduction,
 	}
 	k.EthereumEventProcessor = EthereumEventProcessor{
-		keeper:     k,
 		bankKeeper: bankKeeper,
 	}
```
Comment (fedekunze) [inline module/x/gravity/keeper/keeper.go:71]: I don't really understand why is this hardcoded... what's the point of having the interface if it's always going to be the same? This should be a constructor parameter

File: module/x/gravity/keeper/keeper.go [hunk 2]
```diff
@@ -27,7 +27,7 @@ import (
 type Keeper struct {
 	StakingKeeper          types.StakingKeeper
 	EthereumEventProcessor interface {
```
Comment (fedekunze) [inline module/x/gravity/keeper/keeper.go:29]: this should be defined outside the Keeper struct

File: module/x/gravity/keeper/keeper.go [hunk 3]
```diff
@@ -27,7 +27,7 @@ import (
...
-		Handle(sdk.Context, types.EthereumEvent) error
+		Handle(sdk.Context, *Keeper, types.EthereumEvent) error
```
Comment (fedekunze) [inline module/x/gravity/keeper/keeper.go:30]: imo this should be a `HandleFn` instead of an interface

Comment (yihuang) [reply module/x/gravity/keeper/keeper.go]: I just changed it into a keeper method.

File: module/x/gravity/keeper/keeper.go (hunk 2 above)
Comment (yihuang) [reply module/x/gravity/keeper/keeper.go]: I just changed it into a keeper method.

File: module/x/gravity/keeper/keeper.go (hunk 1 above)
Comment (yihuang) [reply module/x/gravity/keeper/keeper.go]: I just changed it into a keeper method.

**Comment by yihuang** [comment]
> @jackzampolin @levicook is there a reason we have this EthereumEventProcessor type? Why isn't this just a keeper method?

I just changed it into a keeper method.

**Comment by zmanian** [comment]
Can you fix the build error in 
`x/gravity/keeper/ethereum_event_handler_test.go:12:9: undefined: EthereumEventProcessor`

**Comment by yihuang** [comment]
> Can you fix the build error in
> `x/gravity/keeper/ethereum_event_handler_test.go:12:9: undefined: EthereumEventProcessor`

fixed

**Pull request merged**